lib/googlesheets/archivetogooglesheets.go

```diff
@@ -116,6 +116,48 @@ func EnsureMonthColumnExists(month, year string, sheetsData SheetsData) (int, er
 	return chosenColumn, err
 }
 
+// This returns a row number (0-indexed) and a boolean as to whether a row needs to be inserted.
+//  It begins by converting the first column of the rows to string values.
+//  If there are no cells in that column, it just returns (0, false).
+//  If it comes to a cell that is empty or matches the checkName value, it returns the
+//    corresponding row number and false.
+//  If it comes to a cell that has a value greater (alphabetically) than the checkName, it
+//    returns the corresponding row number and true (i.e. a row needs to be inserted)
+func findRowPositionAndWhetherToInsertARow(checkName string, rows [][]interface{}) (int, bool) {
```

Comment:
- Would `any` work here?
- Return int64 to avoid casting it later? (will need to cast it a number of times in function though, so might not be worth it)

```suggestion
func findRowPositionAndWhetherToInsertARow(checkName string, rows [][]any) (int64, bool) {
```